engine/script/src/script_sys.cpp

```diff
@@ -161,13 +161,14 @@ union SaveLoadBuffer
         if (!file)
         {
             Sys_FreeTableSerializationBuffer(buffer);
-            #if !defined(DM_NO_ERRNO)
-                char errmsg[128] = {};
-                dmStrError(errmsg, sizeof(errmsg), errno);
-                return luaL_error(L, "Could not open the file %s, reason: %s.", tmp_filename, errmsg);
-            #else
-                return luaL_error(L, "Could not open the file %s", tmp_filename);
-            #endif
+
+        #if defined(DM_NO_ERRNO)
+            return luaL_error(L, "Could not open the file %s", tmp_filename);
+        #else
+            char errmsg[128] = {};
+            dmStrError(errmsg, sizeof(errmsg), errno);
+            return luaL_error(L, "Could not open the file %s, reason: %s.", tmp_filename, errmsg);
+        #endif
```

Comment:
I flipped the double negativeness of the ifdef

Reply:
`dmStrError()` does the same thing when DM_NO_ERRNO is defined (See dstrings.cpp:236)
That way you don't have to use another `ifdef`
I would recommend removing the "if defined(DM_NO_ERRNO)"

Reply:
Not sure I understand, in this case and after fopen a few lines above, we want to pass in the errno into the function to actually parse it, and if I remove the ifdef errno would be undefined for the platforms where it isn't supported, right?